Add predictor config into Predictor

yic-e opened the pull request:
Summary:
This is the first step of refactoring the Predictor. In this diff the config struct
is introduced and the internal data structure of Predictor has been updated.

Differential Revision: D8843262

fishbone commented on caffe2/core/predictor_config.h:
```diff
@@ -0,0 +1,36 @@
+ * Parameters for a Predictor provided by name.
+ * They are stored as shared_ptr to accommodate parameter sharing
+ */
+using Parameters = std::map<std::string, std::shared_ptr<Blob>>;
```
Should ModelParameters a better name? I feel the PredictorConfig actually is the parameters for the predictor. This field looks more related with the underline model itself.

salexspb replied:
well, these parameters would be for the specific predictor. We don't share the whole map between different predictors, we share blobs. And maps are reconstructed. Also grepping would be simpler with Predictor in the name :)

Commit:
Add predictor config into Predictor (#9434)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/9434

This is the first step of refactoring the Predictor. In this diff the config struct
is introduced and the internal data structure of Predictor has been updated.

Differential Revision: D8843262

fbshipit-source-id: bb6eefcc71b9f5e34caae96005d422a2bcbcc1ea
